app/api/Torrent.js

```diff
@@ -48,8 +48,21 @@ export default class Torrent {
 
     this.engine.add(magnetURI, { path: cacheLocation }, torrent => {
       const server = torrent.createServer();
-      server.listen(port);
-      this.server = server;
+      const thisRef = this;
```

Comment:
`thisRef` is a reference to `this`, as the value of `this` is different when inside the .then() block.